### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by MarissaShieh**
@aku2nnya could you code review this? Thanks!

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -50,28 +58,43 @@ class App extends React.Component {
...
+  }
+
+  newPage(pageNum) {
+    this.setState({
+      pageNum,
     });
   }
 
   render() {
+    const { searchTerm, reviews, allReviews, ratings, foundAverage, avgRating, pageNum, reviewsPerPage, maxPage } = this.state;
```
Comment (aku2nnya) [inline client/components/App.jsx:79]: I should bundle my object desctructuring too...

File: client/components/App.test.jsx [hunk 2]
```diff
@@ -4,44 +4,40 @@ import App from './App.jsx';
 import { doesNotReject } from 'assert';
 
 describe('App component rendering', () => {
+  const wrapper = shallow(<App />);
```
Comment (aku2nnya) [inline client/components/App.test.jsx:7]: nice refactor

File: client/components/Component.module.css [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  justify-content: space-between;
+}
+
+@media only screen and (max-width: 600px) {
+  .totalAndSearch {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+  }
+}
```
Comment (aku2nnya) [inline client/components/Component.module.css]: Please add new linebreak to meet the style guide

File: client/components/PageNavBtns.module.css [hunk 4]
```diff
@@ -0,0 +1,72 @@
+.PageNavBtns{
+  line-height: 10px;
+  display: flex;
+  align-items: center;
+  flex-direction: row;
```
Comment (aku2nnya) [inline client/components/PageNavBtns.module.css]: Believe this is the default flex-direction setting and is unnecessary?

File: client/components/PageNavBtns.module.css [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  line-height: 32px;
+  border-radius: 50%;
+  border: none;
+  color:#008489;
+  height: 30px;
+  width: 30px;
+  padding: 0px 10px;
+  margin: 8px;
+  display: inline-flex; 
+  flex-direction: row;
```
Comment (aku2nnya) [inline client/components/PageNavBtns.module.css]: Not sure if the inline-flex default is also row for flex-direction, but maybe?

File: client/components/PageNavBtns.module.css [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+.elipseBtn {
+  font-size: 16px;
+  text-align: center;
+  line-height: 32px;
+  height: 30px;
+  width: 30px;
+  padding: 0px 10px;
+  margin: 8px;
+  display: inline-flex; 
+  flex-direction: row;
```
Comment (aku2nnya) [inline client/components/PageNavBtns.module.css]: Here too

File: client/components/Ratings.jsx [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
-    let accuracy = 0;
-    let communication = 0;
-    let cleanliness = 0;
-    let location = 0;
-    let checkIn = 0;
-    let value = 0;
+  if(!foundAverage && reviews.length > 0) {  
+    const numReviews = reviews.length;
+    let accuracy, communication, cleanliness, location, checkIn, value;
+    accuracy = communication = cleanliness = location = checkIn = value = 0;
```
Comment (aku2nnya) [inline client/components/Ratings.jsx:13]: Did not know you can do this...

File: client/components/Ratings.jsx [hunk 8]
```diff
@@ -34,17 +34,21 @@ function Ratings(props) {
...
-    <div>
-      <Rating key="accuracyRating" type="accuracy" value={props.ratings.accuracy} />
-      <Rating key="communicationRating" type="communication" value={props.ratings.communication} />
-      <Rating key="cleanlinessRating" type="cleanliness" value={props.ratings.cleanliness} />
-      <Rating key="locationRating" type="location" value={props.ratings.location} />
-      <Rating key="checkInRating" type="checkIn" value={props.ratings.checkIn} />
-      <Rating key="valueRating" type="value" value={props.ratings.value} />
+    <div className='ratingsContainer'>
+      <span className='ratingTriple ratingsLeft'>
+        <Rating key="accuracyRating" type="Accuracy" value={ratings.accuracy} />
```
Comment (aku2nnya) [inline client/components/Ratings.jsx:43]: What are these custom "type" attributes for?

File: client/components/Ratings.module.css [hunk 9]
```diff
@@ -0,0 +1,40 @@
+.ratingsContainer{
+  display: flex;
+  width: 100%;
+  flex-wrap: row wrap;
+  flex-direction: row;
```
Comment (aku2nnya) [inline client/components/Ratings.module.css:5]: I think you would only need 'flex-wrap: wrap' and can remove both row and the whole flex-direction to accomplish the same results.

File: client/components/Review.jsx [hunk 10]
```diff
@@ -1,18 +1,68 @@
...
+  replaceWithBold(paragraph) {
+    return paragraph.replace(this.props.searchTerm, `<b>${this.props.searchTerm}</b>`);
+  }
+
+  render() {
+    const {review} = this.props;
+    let reviewText;
+    if(this.state.expanded === true) {
+      reviewText = review.review_text.split('\n').map( (paragraph, index) => {
+        return <p className='reviewParagraph' key={index} dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(this.replaceWithBold(paragraph))}}></p>;
```
Comment (aku2nnya) [inline client/components/Review.jsx]: What does this do?

File: client/components/Review.module.css [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  letter-spacing: normal;
+  text-align: start;
+  width: 588px;
+  padding: 25px 0px;
+  /* height: 155px; */
+}
+
+.userInfo {
+  display: flex;
+  flex-direction: row;
```
Comment (aku2nnya) [inline client/components/Review.module.css:11]: row should be the default setting, so don't think this is needed

File: client/components/Review.module.css [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+  font-size: 14px;
+}
+
+.reviewParagraph {
+  font-size: 16px;
+  color: #484848;
+  display: inline;
+}
+
+a{
```
Comment (aku2nnya) [inline client/components/Review.module.css:41]: Space between curly brace?

File: client/components/Review.module.css [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  color: #008489;
+  display: inline;
+}
+
+a:hover{
```
Comment (aku2nnya) [inline client/components/Review.module.css:46]: Space between curly brace?

File: client/components/Search.jsx [hunk 14]
```diff
@@ -26,10 +25,23 @@ class Search extends React.Component {
...
+
   render() {
+    let x;
+    if(this.state.searchTerm !== '') {
+      x = <i className="fas fa-times" onClick={()=>{this.clear()}}></i>;
+    }
     return (
-      <form onSubmit={(e) => { this.search(e); }}>
-        <input type='text' placeholder="Search reviews" onChange={(e)=>{this.input(e)}}></input>
+      <form onSubmit={(e) => { this.search(e);}} className='search' >
```
Comment (aku2nnya) [inline client/components/Search.jsx]: Spacing between curly braces

File: client/components/Search.module.css [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+  left: 7px;
+  font-size: 16px;
+}
+
+.fa-times {
+  position: absolute;
+  top: 20px;
+  left: 180px;
+  font-size: 16px;
+}
```
Comment (aku2nnya) [inline client/components/Search.module.css]: Please add new linebreak

File: client/components/SearchStats.module.css [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+  font-size: 14px;
+}
+
+.backToReviews{
+  color: #008489;
+}
+
+.backToReviews:hover{
+  text-decoration: underline;
+}
```
Comment (aku2nnya) [inline client/components/SearchStats.module.css]: Please add new linebreak

File: client/components/SearchStats.test.jsx [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+  location: { type: 'Buffer', data: [4] },
+  value: { type: 'Buffer', data: [3] },
+}];
+
+describe('SearchStats', () => {
+  it('should render w/o errors', () => {
+    const wrapper = shallow(<SearchStats reviews={data} />);
+    expect(wrapper.find('div').exists()).toBe(true);
+  });
+});
```
Comment (aku2nnya) [inline client/components/SearchStats.test.jsx]: Please add new linebreak

File: client/components/Stars.module.css [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+.stars-10:after{width:10%;}
+.stars-20:after{width:20%;}
+.stars-30:after{width:30%;}
+.stars-40:after{width:40%;}
+.stars-50:after{width:50%;}
+.stars-60:after{width:60%;}
+.stars-70:after{width:70%;}
+.stars-80:after{width:90%;}
+.stars-90:after{width:90%;}
+.stars-100:after{width:100%;}
```
Comment (aku2nnya) [inline client/components/Stars.module.css]: Please add new linebreak

**Review by aku2nnya — COMMENTED** [review]
@MarissaShieh Just minor syntax style adjustments, the rest is good to go.

File: client/components/Ratings.jsx (hunk 7 above)
Comment (MarissaShieh) [reply client/components/Ratings.jsx:13]: I didn't either. Ram suggested it to me in a previous code review!

**Pull request merged**